lib/util.js

```diff
@@ -1,40 +1,58 @@
 "use strict";
 /* eslint-disable global-require */
+const isValidObject = require("./is-valid-object");
 
 const HAPI16 = 16;
 const HAPI17 = 17;
+const HAPI18 = 18;
 
 function hapiMajor() {
   let hapiPkg;
   try {
-    hapiPkg = require("hapi/package");
+    hapiPkg = require("@hapi/hapi/package");
   } catch (err) {
-    // ignore
+    hapiPkg = require("hapi/package");
   }
-  return hapiPkg ? +hapiPkg.version.split(".")[0] : HAPI16;
+
+  if (isValidObject(hapiPkg) && hapiPkg.version) {
+    return +hapiPkg.version.split(".")[0];
+  }
+
+  return HAPI16;
 }
-let _isHapi17 = hapiMajor() >= HAPI17;
+
+const hapiV = hapiMajor();
+let _isHapi17 = hapiV >= HAPI17;
+let _isHapi18 = hapiV === HAPI18;
 
 function isHapi17() {
   return _isHapi17;
 }
 
+function isHapi18() {
+  return _isHapi18;
+}
+
 function _testSetHapi17(isHapi17Flag) {
   _isHapi17 = isHapi17Flag;
 }
 
+function _testSetHapi18(isHapi18Flag) {
+  _isHapi18 = isHapi18Flag;
+}
+
 /**
  * Detects the installed Hapi version and use the appropriate plugin
  *
- * @param {*} registers: { hapi16, hapi17 } contains registers for hapi16, hapi17
+ * @param {*} registers: { hapi16, hapi17/hapi18 } contains registers for hapi16, hapi17
  * @param {*} pkg : package for a hapi plugin
  * @returns {*}: Hapi plugin appropriate for installed version
  *
  */
 function universalHapiPlugin(registers, pkg) {
-  if (isHapi17()) {
+  if (_isHapi17 || _isHapi18) {
     return {
-      register: registers.hapi17,
+      register: _isHapi18 ? registers.hapi18 : registers.hapi17,
```

Comment:
are we expecting users to specify hapi17, hapi18, and hapi16?

Reply:
`hapi17` needs to continue to work for both 17, 18, 19, 20, ...., until they change and break the plugin interface, but we can add a new field `hapi17OrUp`.

Reply:
please remove `hapi18`

Reply:
I want to specify only for hapi16 and if 17 and above it has to be simply `registers.hapi`
kinda default.

Reply:
that's actually a good idea, except it would not look very clear on user's side:

```js
module.exports = universalHapiPlugin({
  hapi16: legacyPlugin,
  hapi: otherPlugin
})
```

vs: 

```js
module.exports = universalHapiPlugin({
  hapi16: legacyPlugin,
  hapi17OrUp: otherPlugin
})
```

Reply:
in any case, `hapi17` needs to continue to work, else it's a breaking change.

Reply:
@jchip, do you want me to replace registers.hapi18 with registers. hapi17OrUp ?
Or simply should i remove .hapi18 and use the current .hapi17 for the plugins ?